test/support/assertions.rb

```diff
@@ -9,14 +9,26 @@ def assert_line_num(expected)
       })
     end
 
-    def assert_line_text(expected)
+    def assert_line_text(expected, inverse: false)
       @queue.push(Proc.new {
         result = @last_backlog[1..].join
         expected = Regexp.escape(expected) if expected.is_a?(String)
         msg = "Expected to include `#{expected}` in\n(\n#{result})\n"
-        assert_block(MessageCreationController.new { create_message(msg) }) { result.match? expected }
+        msg = msg.sub("to", "not to") if inverse
+
+        assert_block(MessageCreationController.new { create_message(msg) }) do
+          if inverse
+            !result.match? expected
+          else
+            result.match? expected
+          end
+        end
```

Comment:
Can we separate `assert_block` method and `assert_no_line_text`?

There are 2 reason.
1. Time complexity
`assert_block` method will be executed more than `assert_no_line_text`. Every time `assert_block` method are executed, it have to execute conditional branch two times. This process is a bit CPU-intensive because this method is called a lot.

2. Visual complexity
It's a little hard to understand the codes. test-unit also separates `assert_equal` and `assert_not_equal`
https://github.com/test-unit/test-unit/blob/master/lib/test/unit/assertions.rb#L224
https://github.com/test-unit/test-unit/blob/master/lib/test/unit/assertions.rb#L739

Reply:
> Time complexity

evaluating a boolean in if condition is extremely cheap. `debug_print` evals a boolean value a lot more frequently than the `assert_block`. so tbh I don't think this is a sufficient reason for change.

> Visual complexity

ok I'll change it.

Reply:
Thanks.